hudi-common/src/main/java/org/apache/hudi/common/table/log/block/HoodieAvroDataBlock.java

```diff
@@ -194,6 +185,13 @@ private RecordIterator(Schema readerSchema, Schema writerSchema, byte[] content,
     public static RecordIterator getInstance(HoodieAvroDataBlock dataBlock, byte[] content, InternalSchema internalSchema) throws IOException {
       // Get schema from the header
       Schema writerSchema = new Schema.Parser().parse(dataBlock.getLogBlockHeader().get(HeaderMetadataType.SCHEMA));
+      if (!internalSchema.isEmptySchema()) {
```

Comment:
Why did this block move?

Reply:
I just want to change the readerSchema if we has InternalSchema. DataBlock.getSchema should return the real readerSchema.

Reply:
Ok. Do you think we should merge https://github.com/apache/hudi/pull/6358 first and then rebase rfc-46 on master?

Reply:
@wzx140 yes, that's the plan. #6358 will be merged today.